uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java

```diff
@@ -0,0 +1,39 @@
+package com.aws.greengrass.steps;
+
+import io.cucumber.guice.ScenarioScoped;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.When;
+
+@ScenarioScoped
+public class MqttControlSteps {
+
+    @And("the control component configured:")
+    public void configureControl(String configurationSpec){
+        //@TODO Implement method
+    }
+
+    @When("Client {word} subscribe to {word} qos {int}")
```

Comment:
These operation do with MQTT connection, not with MQTT client itself.
For example: we can run the same MQTT client  based on  'AWS IoT device SDK for java v2' on Linux as agentLin and on Windows as agentWin. One of them can establish connection with IoT Core directly and another to EMQX  MQTT broker of Greengrass. Currently to do operation like subscribe or publish we need know 1. agentID (string) 2. connectionId (integer) inside agent.

Maybe to simplify we can create mapping of pair {agentId, connectionId} to single {logicaConnectionId}

Reply:
Looks like we need to define some naming conventions for the project.
When we compile and run source code from  'AWS IoT device SDK for java v2' on Linux is starts OS process. In real world scenarios this client code is only a dependency in embedded device application, right?
In GGAD scenario they have abbreviation GGAD(Greengrass Aware Device). 
Also such embedded applications connect to Core device as a Client devices (AWS console naming)
So I guess that we need to chose between:
clientId, deviceId, clientDeviceId, ggadId
Not sure that we can use agentId, to avoid misunderstanding. There are several GG components that are "agents", like AWS Systems Manager Agent
Please, dont worry about connectionId. We are free to overload any test step, something like:
Client {clientId} subscribe ... - if we plan to use single connection
Client {clientId}, connection {connectionId} subscribe ... - when we want to test ability to use multiple connections
What do you think?

Reply:
fixed 7bec4aab4bf10f182d29bef87204a704dac3bdce